tests/osaka/eip7883_modexp_gas_increase/conftest.py

```diff
@@ -40,12 +40,38 @@ def call_contract_post_storage() -> Storage:
 
 
 @pytest.fixture
-def call_succeeds() -> bool:
+def total_tx_gas_needed(
+    fork: Fork, modexp_expected: bytes, modexp_input: ModExpInput, precompile_gas: int
+) -> int:
+    """Calculate total tx gas needed for the transaction."""
+    intrinsic_gas_cost_calculator = fork.transaction_intrinsic_cost_calculator()
+    memory_expansion_gas_calculator = fork.memory_expansion_gas_calculator()
+    sstore_gas = fork.gas_costs().G_STORAGE_SET * (len(modexp_expected) // 32)
+    extra_gas = 100_000
```

Comment:
I think we no longer need `sstore_gas`, since we switch to hash-based comparison method. I also do the refactor in my [PR](https://github.com/ethereum/execution-spec-tests/pull/2098). What do you think?

```suggestion
intrinsic_gas_cost_calculator = fork.transaction_intrinsic_cost_calculator()
memory_expansion_gas_calculator = fork.memory_expansion_gas_calculator()
extra_gas = 500_000
```

Reply:
If this works, also remove the `sstore_gas` in the return.